### ISSUE TITLE
Eval bug: get_logits_ith: invalid logits id 0, reason: no logits

### ISSUE THREAD

**Issue by niutech** [Anbeeld/beellama.cpp#81]
### Name and Version

beellama-v0.3.1
also version: 1 (a620cbd)                                                                                                                                                                                               built with GNU 15.2.0 for Linux x86_64 

### Operating systems

Linux

### GGML backends

CUDA

### Hardware

NVIDIA GeForce RTX 3060

### Models

speakleash/Bielik-11B-v3.0-DFlash + speakleash/Bielik-11B-v3.0-Instruct

### Problem description & steps to reproduce

When I download prebuilt beellama.cpp v0.3.1 for Ubntu x64 with CUDA 13.1 and run `./llama-cli -m Bielik-11B-v3.0-Instruct-Q8_0.gguf --spec-type dflash --spec-draft-model Bielik-11B-v3.0-DFlash-Q8_0.gguf --spec-draft-ngl all -fa on -c 2048`, I get a single word of response from the model and then the following error: `get_logits_ith: invalid logits id 0, reason: no logits`.

### Relevant log output

<details>
<summary>Logs</summary>

```console
/__w/beellama.cpp/beellama.cpp/common/sampling.cpp:154: GGML_ASSERT(logits != nullptr) failed
0.04.559.728 E get_logits_ith: invalid logits id 0, reason: no logits
[New LWP 246577]
[New LWP 246574]
[New LWP 246573]
[New LWP 246572]
[New LWP 246571]
[New LWP 246568]
This GDB supports auto-downloading debuginfo from the following URLs:
<https://debuginfod.ubuntu.com>
Enable debuginfod for this session? (y or [n]) [answered N; input not from terminal]
Debuginfod has been disabled.
To make this setting permanent, add 'set debuginfod enabled off' to .gdbinit.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/x86_64-linux-gnu/libthread_db.so.1".
__syscall_cancel_arch () at ../sysdeps/unix/sysv/linux/x86_64/syscall_cancel.S:56
#0  __syscall_cancel_arch () at ../sysdeps/unix/sysv/linux/x86_64/syscall_cancel.S:56
#1  0x00007f4531ea05ec in __internal_syscall_cancel (a1=a1@entry=108162062646212, a2=<optimized out>, a3=a3@entry=2648924688, a4=a4@entry=140731547345808, a5=a5@entry=0, a6=a6@entry=4294967295, nr=202) at ./nptl/cancellation.c:49
#2  0x00007f4531ea09c7 in __futex_abstimed_wait_common64 (private=32581, futex_word=0x625f71527bc4, expected=2648924688, op=32716, abstime=0x7ffe9de36390, cancel=true) at ./nptl/futex-internal.c:57
#3  __futex_abstimed_wait_common (futex_word=0x625f71527bc4, expected=2648924688, clockid=1, abstime=0x7ffe9de36390, private=32581, cancel=true) at ./nptl/futex-internal.c:87
#4  __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x625f71527bc4, expected=expected@entry=3, clockid=clockid@entry=1, abstime=abstime@entry=0x7ffe9de36390, private=private@entry=0) at ./nptl/futex-internal.c:139
#5  0x00007f4531ea3969 in __pthread_cond_wait_common (cond=<optimized out>, mutex=0x625f71527b78, clockid=1, abstime=<optimized out>) at ./nptl/pthread_cond_wait.c:421
#6  ___pthread_cond_clockwait64 (cond=<optimized out>, mutex=0x625f71527b78, clockid=1, abstime=<optimized out>) at ./nptl/pthread_cond_wait.c:517
#7  ___pthread_cond_clockwait64 (cond=<optimized out>, mutex=0x625f71527b78, clockid=1, abstime=<optimized out>) at ./nptl/pthread_cond_wait.c:505
#8  0x00007f4532652012 in server_response::recv_with_timeout(std::unordered_set<int, std::hash<int>, std::equal_to<int>, std::allocator<int> > const&, int) () from /home/xxx/beellama-v0.3.1/libllama-cli-impl.so
#9  0x00007f4532656542 in server_response_reader::next(std::function<bool ()> const&) () from /home/xxx/beellama-v0.3.1/libllama-cli-impl.so
#10 0x00007f4532600a07 in cli_context::generate_completion[abi:cxx11](result_timings&) () from /home/xxx/beellama-v0.3.1/libllama-cli-impl.so
#11 0x00007f45325ef39d in llama_cli(int, char**) () from /home/xxx/beellama-v0.3.1/libllama-cli-impl.so
#12 0x00007f4531e2a601 in __libc_start_call_main (main=main@entry=0x625f4a0ee270 <main>, argc=argc@entry=13, argv=argv@entry=0x7ffe9de3a3f8) at ../sysdeps/nptl/libc_start_call_main.h:59
#13 0x00007f4531e2a718 in __libc_start_main_impl (main=0x625f4a0ee270 <main>, argc=13, argv=0x7ffe9de3a3f8, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffe9de3a3e8) at ../csu/libc-start.c:360
#14 0x0000625f4a0ee2a5 in _start ()
[Inferior 1 (process 246565) detached]
```
</details>

**Comment by deadendroid** [comment]
I think problem is with **--spec-draft-n-min** setting (default to 0)

get_logits_ith: invalid logits id 0, reason: no logits

It looks like it does not generate speculative tokens. 
Take a look at quick start here https://huggingface.co/speakleash/Bielik-11B-v3.0-DFlash
Looks like both SGLANG and VLLM have set this parameter explicitly.

```
export SGLANG_ALLOW_OVERWRITE_LONGER_CONTEXT_LEN=1

python -m sglang.launch_server \
    --model-path speakleash/Bielik-11B-v3.0-Instruct \
    --speculative-algorithm DFLASH \
    --speculative-draft-model-path speakleash/Bielik-11B-v3.0-DFlash \

    --speculative-num-draft-tokens 16
``` 
Anyway. It works with **--spec-draft-n-min 16**
Pozdro.

**Comment by niutech** [comment]
@Anbeeld It works with `--spec-draft-n-min 16`, thanks!
